src/Cli/dotnet/commands/dotnet-new/MSBuildEvaluation/MSBuildEvaluator.cs

```diff
@@ -0,0 +1,201 @@
+
+                //For SDK-style project, if the project was restored "RestoreSuccess" property will be set to true.
+                if (!evaluatedProject.GetProperty("RestoreSuccess")?.EvaluatedValue.Equals("true", StringComparison.OrdinalIgnoreCase) ?? true)
+                {
+                    _logger?.LogDebug("Project is not restored, exiting.");
+                    return MSBuildEvaluationResult.CreateNoRestore(projectPath);
+                }
+
+                //If target framework is set, no further evaluation is needed.
+                if (!string.IsNullOrWhiteSpace(targetFramework))
+                {
+                    _logger?.LogDebug("Project is SDK style, single TFM:{0}, evaluation succeeded.", targetFramework);
+                    return SDKStyleEvaluationResult.CreateSuccess(projectPath, targetFramework, evaluatedProject);
+                }
+
+                //If target framework is not set, then presumably it is multi-target project.
+                //If there are no target frameworks, it is not expected.
+                if (targetFrameworks == null)
+                {
+                    _logger?.LogDebug("Project is SDK style, but does not specify the framework.");
+                    return MSBuildEvaluationResult.CreateFailure(projectPath, string.Format(LocalizableStrings.MSBuildEvaluator_Error_NoTargetFramework, projectPath));
+                }
+
+                //For multi-target project, we need to do additional evaluation for each target framework.
+                Dictionary<string, Project?> evaluatedTfmBasedProjects = new Dictionary<string, Project?>();
+                foreach (string tfm in targetFrameworks)
+                {
+                    _logger?.LogDebug("Evaluating project for target framework: {0}", tfm);
+                    evaluatedTfmBasedProjects[tfm] = RunEvaluate(projectPath, tfm);
+                }
+                _logger?.LogDebug("Project is SDK style, multi-target, evaluation succeeded.");
+                return MultiTargetEvaluationResult.CreateSuccess(projectPath, evaluatedProject, evaluatedTfmBasedProjects);
+
+            }
+            catch (Exception e)
+            {
+                _logger?.LogDebug("Unexpected error: {0}", e.Message);
+                return MSBuildEvaluationResult.CreateFailure(projectPath, e.Message);
+            }
+        }
+
+        private static Project RunEvaluate(string projectToLoad, string? tfm = null)
+        {
+            if (File.Exists(projectToLoad))
+            {
+                ProjectCollection projectCollection = new ProjectCollection();
+
+                if (!string.IsNullOrWhiteSpace(tfm))
+                {
+                    projectCollection.SetGlobalProperty("TargetFramework", tfm);
+                }
+
+                var project = new Project(
+                        projectToLoad,
+                        projectCollection.GlobalProperties,
+                        toolsVersion: "Current",
```

Comment:
We're gradually phasing out the whole concept of `toolsVersion`, so no need to specify it.

```suggestion
                        toolsVersion: null,
```